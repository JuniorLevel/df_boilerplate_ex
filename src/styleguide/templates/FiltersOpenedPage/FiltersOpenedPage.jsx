// @flow

import {
	FiltersClosedListProvider,
	FiltersOpened,
	Layout,
} from '@juniorlevel/df_boilerplate-library';
import React, { useState } from 'react';

export const FiltersOpenedPage = (): React.Node => {
	const buttons = [
		{
			id: 1,
			title: 'Orders',
			onClick: () => {},
		},
		{
			id: 2,
			title: 'Compare',
			onClick: () => {},
		},
		{
			id: 3,
			title: 'Admin',
			onClick: () => {},
		},
	];

	const [filters] = useState([
		{
			id: 1,
			cardItem: [{ id: 1, status: 'changed', text: 'lorem-1' }],
		},

		{
			id: 2,
			cardItem: [
				{ id: 1, status: 'updated', text: 'project-1' },
				{ id: 2, status: 'changed', text: 'project-2' },
				{ id: 3, status: 'updated', text: 'project-3' },
			],
		},
	]);

	return (
		<FiltersClosedListProvider>
			<Layout buttons={buttons} title="Caption" avatar={{ title: 'AV' }}>
				<FiltersOpened isOpen filtersCards={filters} />
			</Layout>
		</FiltersClosedListProvider>
	);
};
