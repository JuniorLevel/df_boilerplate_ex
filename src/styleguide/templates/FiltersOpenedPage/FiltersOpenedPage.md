```jsx
import React, { useState } from 'react';
import {
	FiltersClosedListProvider,
	FiltersOpened,
	Layout,
} from '@juniorlevel/df_boilerplate-library';

const buttons = [
	{
		id: 1,
		title: 'Orders',
		onClick: () => {},
	},
	{
		id: 2,
		title: 'Compare',
		onClick: () => {},
	},
	{
		id: 3,
		title: 'Admin',
		onClick: () => {},
	},
];

const [filters, setFilters] = useState([
	{
		id: 1,
		cardItem: [{ id: 1, status: 'changed', text: 'lorem-1' }],
	},

	{
		id: 2,
		cardItem: [
			{ id: 1, status: 'updated', text: 'project-1' },
			{ id: 2, status: 'changed', text: 'project-2' },
			{ id: 3, status: 'updated', text: 'project-3' },
		],
	},
]);

<FiltersClosedListProvider>
	<Layout buttons={buttons} title="Caption" avatar={{ title: 'AV' }}>
		<FiltersOpened isOpen={true} filtersCards={filters} />
	</Layout>
</FiltersClosedListProvider>;
```

```jsx noeditor
import Image1 from '../../assets/FiltersOpened/filters-opened1.png';
import Image2 from '../../assets/FiltersOpened/filters-opened2.png';
<div>
	<img src={Image1} />
	<img src={Image2} />
</div>;
```
