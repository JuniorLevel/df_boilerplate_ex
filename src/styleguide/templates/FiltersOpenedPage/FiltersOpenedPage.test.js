import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { FiltersOpenedPage } from './FiltersOpenedPage';
import '../../../../__mocks__/fileMock';

describe('FiltersOpenedPage', () => {
	test('should render FiltersOpenedPage component without crashing', () => {
		const { container } = render(<FiltersOpenedPage />);
		expect(container).toBeInTheDocument();
	});
	test('matches the snapshot desktop', () => {
		const { container: desktop } = render(
			<ResponsiveContext.Provider value={{ width: 1920 }}>
				<BrowserRouter>
					<FiltersOpenedPage />
				</BrowserRouter>
			</ResponsiveContext.Provider>
		);
		expect(desktop).toMatchSnapshot();
	});
	test('matches the snapshot laptop', () => {
		const { container: laptop } = render(
			<ResponsiveContext.Provider value={{ width: 1024 }}>
				<BrowserRouter>
					<FiltersOpenedPage />
				</BrowserRouter>
			</ResponsiveContext.Provider>
		);
		expect(laptop).toMatchSnapshot();
	});
	test('matches the snapshot mobile', () => {
		const { container: mobile } = render(
			<ResponsiveContext.Provider value={{ width: 360 }}>
				<BrowserRouter>
					<FiltersOpenedPage />
				</BrowserRouter>
			</ResponsiveContext.Provider>
		);
		expect(mobile).toMatchSnapshot();
	});
});
