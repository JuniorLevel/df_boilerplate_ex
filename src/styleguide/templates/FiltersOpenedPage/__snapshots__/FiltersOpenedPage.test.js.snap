// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersOpenedPage matches the snapshot desktop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-z125ah-0 ydqdt css-1c0na6j"
  >
    <div
      class="ant-layout css-1c0na6j"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-z125ah-1 MFHYS css-1c0na6j"
      />
    </div>
  </div>
</div>
`;

exports[`FiltersOpenedPage matches the snapshot laptop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-z125ah-0 ydqdt css-1c0na6j"
  >
    <div
      class="ant-layout css-1c0na6j"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-z125ah-1 MFHYS css-1c0na6j"
      />
    </div>
  </div>
</div>
`;

exports[`FiltersOpenedPage matches the snapshot mobile 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-z125ah-0 ydqdt css-1c0na6j"
  >
    <div
      class="ant-layout css-1c0na6j"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-z125ah-1 MFHYS css-1c0na6j"
      />
    </div>
  </div>
</div>
`;
